web/packages/teleterm/src/ui/services/deepLinks/deepLinksService.ts

```diff
@@ -88,13 +88,70 @@ export class DeepLinksService {
         break;
       }
       case '/authenticate_web_device': {
-        // TODO (avatus): add authenticate device web confirmation dialog
-        // this case is a stub and will not be reached
+        const id = result.url?.searchParams?.id;
+        const token = result.url?.searchParams?.token;
+        await this.askAuthorizeDeviceTrust(result.url, id, token);
         break;
       }
     }
   }
 
+  private async askAuthorizeDeviceTrust(
+    url: DeepURL,
+    id: string,
+    token: string
+  ): Promise<void> {
+    const rootClusterId = url.hostname;
+    const clusterAddress = url.host;
+    const prefill = {
+      clusterAddress,
+      username: url.username,
+    };
+
+    const rootClusterUri = routing.getClusterUri({ rootClusterId });
+    const rootCluster = this.clustersService.findCluster(rootClusterUri);
+
+    if (!rootCluster) {
+      const { canceled } = await new Promise<{ canceled: boolean }>(resolve => {
+        this.modalsService.openRegularDialog({
+          kind: 'cluster-connect',
+          clusterUri: undefined,
+          reason: undefined,
+          prefill,
+          onSuccess: () => resolve({ canceled: false }),
+          onCancel: () => resolve({ canceled: true }),
+        });
+      });
+
+      if (canceled) {
+        return;
+      }
+    }
+
+    const { isAtDesiredWorkspace } =
+      await this.workspacesService.setActiveWorkspace(rootClusterUri, prefill);
+
+    if (!isAtDesiredWorkspace) {
+      return;
+    }
+
+    this.modalsService.openImportantDialog({
+      kind: 'device-trust-authorize',
+      onCancel: () => console.log('cancelled'),
+      onConfirm: async () => {
+        await this.clustersService.authenticateWebDevice(rootClusterUri, {
+          id,
+          token,
+        });
+        // open url to confirm the token. This endpoint verifies the token and "upgrades"
+        // the web session and redirects to "/web"
+        window.open(
```

Comment:
This reminds me, let's remember to double check this later if `window.open` here with `url.host`/`rootCluster.proxyHost` works as expected.

When working on app access and working on the allow list for proxy hosts, I ran into a weird case where clicking on `example.com:443` would report the URL as `example.com`. That's why when saving the proxy list, I strip the default port number:

https://github.com/gravitational/teleport/blob/d56ed2c5f88def3d8d0313684c7f867d44d09a65/web/packages/teleterm/src/services/tshd/cluster.ts#L21-L37

https://github.com/gravitational/teleport/blob/d56ed2c5f88def3d8d0313684c7f867d44d09a65/web/packages/teleterm/src/mainProcess/rootClusterProxyHostAllowList.ts#L86

Fingers crossed that `window.open` has the same effect as clicking a link and there's no extra work that needs to be done here.

Reply:
Checked this and everything seems to work fine with/without the port